apps/ledger-live-desktop/src/newArch/Collectibles/hooks/useNftDetailDrawer.tsx

```diff
@@ -0,0 +1,116 @@
+import { Account, ProtoNFT } from "@ledgerhq/types-live";
+import { useDispatch, useSelector } from "react-redux";
+import { getNFTById } from "~/renderer/reducers/accounts";
+import { State } from "~/renderer/reducers";
+import { useNftMetadata, useNftCollectionMetadata } from "@ledgerhq/live-nft-react";
+import { getFloorPrice } from "@ledgerhq/live-nft/api/metadataservice";
+import { getCryptoCurrencyById } from "@ledgerhq/live-common/currencies/index";
+import { FloorPrice, NFTMetadata } from "@ledgerhq/types-live";
+import { getMetadataMediaType } from "~/helpers/nft";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { openModal } from "~/renderer/actions/modals";
+import { createDetails } from "LLD/Collectibles/utils/createNftDetailsArrays";
+import { setDrawer } from "~/renderer/drawers/Provider";
+import isEmpty from "lodash/isEmpty";
+
+const useNftDetailDrawer = (account: Account, nftId: string) => {
+  const dispatch = useDispatch();
+  const state = useSelector((state: State) => state);
+
+  const protoNft = useMemo(() => {
+    return (
+      getNFTById(state, {
+        nftId,
+      }) || ({} as ProtoNFT)
+    );
+  }, [state, nftId]);
+
+  const doNotOpenDrawer = isEmpty(protoNft);
+
+  const { status: collectionStatus, metadata: collectionMetadata } = useNftCollectionMetadata(
+    protoNft.contract,
+    protoNft.currencyId,
+  );
+
+  const { status: nftStatus, metadata } = useNftMetadata(
+    protoNft.contract,
+    protoNft.tokenId,
+    protoNft.currencyId,
+  );
+
+  const isLoading = useMemo(
+    () => nftStatus === "loading" || collectionStatus === "loading",
+    [collectionStatus, nftStatus],
+  );
+
+  const [floorPriceLoading, setFloorPriceLoading] = useState(false);
+  const [ticker, setTicker] = useState("");
+  const [floorPrice, setFloorPrice] = useState<number | null>(null);
+  const currency = useMemo(() => getCryptoCurrencyById(protoNft.currencyId), [protoNft.currencyId]);
+
+  useEffect(() => {
+    setFloorPriceLoading(true);
+    getFloorPrice(protoNft, currency)
+      .then((result: FloorPrice | null) => {
+        if (result) {
+          setTicker(result.ticker);
+          setFloorPrice(result.value);
+        }
+      })
+      .finally(() => setFloorPriceLoading(false));
+  }, [protoNft, currency]);
```

Comment:
You can try to use TanstackQuery to handle that :) You have example in Market or for NFTSpanFilter.

You will not have to handle Loading States 

Don't hesitate to ask me if you need help :)